src/main/webapp/app/exercises/programming/manage/grading/feedback-analysis/Modal/feedback-filter-modal.component.html

```diff
@@ -0,0 +1,74 @@
+<form [formGroup]="filterForm" (ngSubmit)="applyFilter()">
+    <div class="modal-header">
+        <h5 class="modal-title" jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.modalTitle"></h5>
+        <button type="button" class="btn-close" (click)="closeModal()"></button>
+    </div>
+    <div class="modal-body">
+        <div class="form-group">
+            <label jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.task"></label>
+            <div class="row">
+                @for (task of generateTaskArray(); track task) {
+                    <div class="col-4">
+                        <div class="form-check">
+                            <input
+                                type="checkbox"
+                                class="form-check-input"
+                                id="task-{{ task }}"
+                                [value]="task"
+                                (change)="onCheckboxChange($event, 'tasks')"
+                                [checked]="filterForm.value.tasks.includes(task.toString())"
```

Comment:
Could we track this state without having a function call in the template? Maybe with a record?

Reply:
i added on push detection strategy to avoid this for now. everything else did not work or caused issues with signals